File: say.py
# Say-Salomon

import json
import requests

# Function to convert text to speech
def text_to_speech(text):
    headers = {"Authorization": "Bearer 🔑 Your_API_Key"}
    url = "https://api.edenai.run/v2/audio/text_to_speech"
    payload = {"providers": "google,amazon", "language": "en-US", "option": "MALE", "text": text}
    response = requests.post(url, json=payload, headers=headers)
    result = json.loads(response.text)
    return result['google']['audio']

# Function to convert text to speech asynchronously
def text_to_speech_async(text):
    headers = {"Authorization": "Bearer 🔑 Your_API_Key"}
    url = "https://api.edenai.run/v2/audio/text_to_speech_async"
    payload = {"providers": "lovoai", "language": "en-US", "option": "MALE", "text": text}
    response = requests.post(url, json=payload, headers=headers)
    result = json.loads(response.text)
    return result

# Function to translate text
def translate_text(text, source_language, target_language):
    headers = {"Authorization": "Bearer 🔑 Your_API_Key"}
    url = "https://api.edenai.run/v2/translation/automatic_translation"
    payload = {"providers": "google,amazon", 'source_language': source_language, 'target_language': target_language, 'text': text}
    response = requests.post(url, json=payload, headers=headers)
    result = json.loads(response.text)
    return result['google']['text']

# Function to transcribe speech to text asynchronously
def speech_to_text_async(file_url):
    headers = {"Authorization": "Bearer 🔑 Your_API_Key"}
    url = "https://api.edenai.run/v2/audio/speech_to_text_async"
    payload = {"providers": "google, amazon", "language": "en-US", "file_url": file_url}
    response = requests.post(url, json=payload, headers=headers)
    result = json.loads(response.text)
    return result

# Example usage
text = "This is a test"
audio = text_to_speech(text)
print(audio)

translated_text = translate_text(text, "en", "fr")
print(translated_text)

file_url = "🔗 URL of your audio"
transcription = speech_to_text_async(file_url)
print(transcription)

File: say.rs
```rust
use reqwest::header::{HeaderMap, AUTHORIZATION};
use serde_json::json;

// Function to convert text to speech
async fn text_to_speech(text: &str) -> Result<String, reqwest::Error> {
    let url = "https://api.edenai.run/v2/audio/text_to_speech";
    let payload = json!({
        "providers": "google,amazon",
        "language": "en-US",
        "option": "MALE",
        "text": text
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiMjkzNDQyMDUtOGY1ZC00YzcyLTg5NjctYTE4Mjc5Njc4ODRlIiwidHlwZSI6ImFwaV90b2tlbiJ9.9XRl_reCsr398oL0bA2_BnoS6AA98la7QV8lrh5k25o")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result["google"]["audio"].to_string())
}

// Function to convert text to speech asynchronously
async fn text_to_speech_async(text: &str) -> Result<serde_json::Value, reqwest::Error> {
    let url = "https://api.edenai.run/v2/audio/text_to_speech_async";
    let payload = json!({
        "providers": "lovoai",
        "language": "en-US",
        "option": "MALE",
        "text": text
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiMjkzNDQyMDUtOGY1ZC00YzcyLTg5NjctYTE4Mjc5Njc4ODRlIiwidHlwZSI6ImFwaV90b2tlbiJ9.9XRl_reCsr398oL0bA2_BnoS6AA98la7QV8lrh5k25o")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result)
}

// Function for automatic translation
async fn automatic_translation(
    text: &str,
    source_language: &str,
    target_language: &str,
) -> Result<String, reqwest::Error> {
    let url = "https://api.edenai.run/v2/translation/automatic_translation";
    let payload = json!({
        "providers": "google,amazon",
        "source_language": source_language,
        "target_language": target_language,
        "text": text
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer 🔑 Your_API_Key")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result["google"]["text"].to_string())
}

// Function for speech to text conversion
async fn speech_to_text_async(file_url: &str) -> Result<serde_json::Value, reqwest::Error> {
    let url = "https://api.edenai.run/v2/audio/speech_to_text_async";
    let payload = json!({
        "providers": "google, amazon",
        "language": "en-US",
        "file_url": file_url
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer 🔑 Your_API_Key")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result)
}

// Function for text generation
async fn generate_text() -> Result<String, reqwest::Error> {
    let url = "https://api.edenai.run/v2/text/generation";
    let payload = json!({
        "providers": "openai,cohere",
        "text": "this is a test of Eden AI",
        "temperature": 0.2,
        "max_tokens": 250
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer 🔑 Your_API_Key")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result["openai"]["generated_text"].to_string())
}

// Function for spell check
async fn spell_check(text: &str) -> Result<serde_json::Value, reqwest::Error> {
    let url = "https://api.edenai.run/v2/text/spell_check";
    let payload = json!({
        "providers": "openai,microsoft",
        "language": "en",
        "text": text
    });

    let client = reqwest::Client::new();
    let response = client
        .post(url)
        .json(&payload)
        .header(AUTHORIZATION, "Bearer 🔑 Your_API_Key")
        .send()
        .await?;

    let result = response.json::<serde_json::Value>().await?;
    Ok(result["openai"]["items"].to_owned())
}

#[tokio::main]
async fn main() {
    let text = "Hello, how are you?";

    if let Ok(audio) = text_to_speech(text).await {
        println!("{}", audio);
    }

    if let Ok(translated_text) = automatic_translation(text, "en", "fr").await {
        println!("{}", translated_text);
    }

    let file_url = "🔗 URL of your audio file";
    if let Ok(transcribed_text) = speech_to_text_async(file_url).await {
        println!("{}", transcribed_text);
    }

    if let Ok(generated_text) = generate_text().await {
        println!("{}", generated_text);
    }

    if let Ok(spell_checked_text) = spell_check(text).await {
        println!("{}", spell_checked_text);
    }
}
```
